src/libraries/System.Text.RegularExpressions/tests/Regex.Split.Tests.cs

```diff
@@ -57,7 +57,59 @@ public static IEnumerable<object[]> Split_TestData()
+            yield return new object[] { "a(.)c(.)e", "123abcde456aBCDe789", RegexOptions.ExplicitCapture, 19, 0, new string[] { "123", "456aBCDe789" } };
+            yield return new object[] { "a.c.e", "123abcde456aBCDe789", RegexHelpers.RegexOptionNonBacktracking, 19, 0, new string[] { "123", "456aBCDe789" } };
+            yield return new object[] { "a.c.e", "123abcde456aBCDe789", RegexOptions.ExplicitCapture, 19, 0, new string[] { "123", "456aBCDe789" } };
+            yield return new object[] { "a.c.e", "123abcde456aBCDe789", RegexHelpers.RegexOptionNonBacktracking | RegexOptions.IgnoreCase, 19, 0, new string[] { "123", "456", "789" } };
+            yield return new object[] { "a.c.e", "123abcde456aBCDe789", RegexOptions.ExplicitCapture | RegexOptions.IgnoreCase, 19, 0, new string[] { "123", "456", "789" } };
+
+            // RegexOptions.NonBacktracking also ignores named captures as if they were not given
+            yield return new object[] { "a(?<dot1>.)c(.)e", "123abcde456aBCDe789", RegexHelpers.RegexOptionNonBacktracking, 19, 0, new string[] { "123", "456aBCDe789" } };
+            yield return new object[] { "a(?<dot1>.)c(.)e", "123abcde456aBCDe789", RegexHelpers.RegexOptionNonBacktracking | RegexOptions.IgnoreCase, 19, 0, new string[] { "123", "456", "789" } };
+
+            // IgnoreCase
+            yield return new object[] { "[abc]", "1A2B3C4", RegexHelpers.RegexOptionNonBacktracking | RegexOptions.IgnoreCase, 7, 0, new string[] { "1", "2", "3", "4" } };
+
+            // Use of custom index and count
+            yield return new object[] { ":", "kkk:lll:mmm:nnn:ooo", RegexHelpers.RegexOptionNonBacktracking, 2, 0, new string[] { "kkk", "lll:mmm:nnn:ooo" } };
+            yield return new object[] { ":", "kkk:lll:mmm:nnn:ooo", RegexHelpers.RegexOptionNonBacktracking, 3, 6, new string[] { "kkk:lll", "mmm", "nnn:ooo" } };
+        }
+
+        // Tricky corner cases involving empty matches of anchors
+        public static IEnumerable<object[]> Split_TestData_AnchorMatches()
+        {         
+            foreach (var options in new RegexOptions[] { RegexHelpers.RegexOptionNonBacktracking, RegexOptions.None, RegexOptions.Compiled })
+            {
+                yield return new object[] { @"\b", "Hello World!", options, 3, 6, new string[] { "Hello ", "World", "!" } };
+                yield return new object[] { @"\b", "Hello World!", options, 0, 0, new string[] { "", "Hello", " ", "World", "!" } };
+                yield return new object[] { @"^", "Hello \nWorld!", options | RegexOptions.Multiline, 0, 0, new string[] { "", "Hello \n", "World!" } };
+                yield return new object[] { @"$", "Hello \nWorld!", options | RegexOptions.Multiline, 0, 0, new string[] { "Hello ", "\nWorld!", "" } };
+            }
+        }
+
         [Theory]
+        [MemberData(nameof(Split_TestData_AnchorMatches))]
+        [MemberData(nameof(Split_TestData_NonBacktracking))]
```

Comment:
I'm curious how you're running tests locally?  I do:
```C#
dotnet build /t:test
```
That will run the tests first on .NET Core and then on .NET Framework... I expect this will result in a lot of failures on .NET Framework.

I can keep fixing them up after the fact, but I'm curious why you're not seeing these failures locally.

Reply:
When I run them locally I just right-click on the test project and select Run Tests. So far this has worked well. So I did not know I need to do anything else special.

Reply:
But also, all 8 checks have passed. I would have expected there to be some failures otherwise in the cloud build process.

Reply:
More precisely. I select `Run Tests` from  the context menu of the `System.Text.RegularExpressions.Tests` project and then inspect the Tests output. With the latest updates for example it says:
```
========== Starting test run ==========
[xUnit.net 00:00:00.02] xUnit.net VSTest Adapter v2.4.2+2d84eb3141 (64-bit .NET 7.0.0-dev)
[xUnit.net 00:00:01.01]   Starting:    System.Text.RegularExpressions.Tests (parallel test collections = on, max threads = 56)
[xUnit.net 00:00:10.15]   Finished:    System.Text.RegularExpressions.Tests
========== Test run finished: 9168 Tests (9168 Passed, 0 Failed, 0 Skipped) run in 10.2 sec ==========
```
This is the way I have use the solution and run tests so far.

Reply:
So I guess I need to add additional checks that under net48 the test with the new option are not run. So far I've not done that, but will do going forward. I'll see from your updates what needs to be done. 
I do see the failures when I now run `dotnet build /t:test` in the test directory (I did not run test like that before).